### PR TITLE
feat(UI): user profile picture is configurable

### LINKED ISSUE #774: User profile picture should be configurable
### Is your feature request related to a problem? Please describe.

There is no way for users to set or change their profile picture. The app shows initials or a placeholder avatar everywhere (sidebar, nav menu, user management). Users cannot personalize their accounts.

### Describe the solution you'd like

Add a profile picture upload option to the user account page. Store the image (or accept a URL/Gravatar fallback). Display it anywhere the user avatar appears.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by VishnuM049**
## Purpose / Description
Made the user profile picture changeable

## Fixes
* Fixes #774

## Approach
Added avatar_url column to the users table, two new auth endpoints 
-(PUT/DELETE) with base64 validation and magic byte checks to validate if it's an image
-crop dialog using CSS overlays with Canvas only at save time

## How Has This Been Tested?
tested by changing in frontend and editing multiple times

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] All commits are signed off (`git commit -s`) per the [DCO](https://developercertificate.org/)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

image

### CODE REVIEW

**Comment by ShaanNarendran** [comment]
@VishnuM049 did u test all the edge cases like different file types, sizes etc?

**Comment by VishnuM049** [comment]
> @VishnuM049 did u test all the edge cases like different file types, sizes etc?

tested file types, sizes, fixed login/logout persistence, tested the selection square and removal

**Comment by Haz3-jolt** [comment]
Hi! We recently added a CLA requirement to the repo. Could you rebase your branch on main and force push? That will trigger the CLA bot to comment with the signing link. Thanks!

File: observal-server/alembic/versions/0003_add_avatar_url.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""Add users.avatar_url TEXT column for user profile pictures
+
+Supports base64 data URLs for user-uploaded profile pictures. Nullable to
+allow users without avatars (shows initials instead).
+
+Revision ID: 0003
```
Comment (Haz3-jolt) [inline observal-server/alembic/versions/0003_add_avatar_url.py]: 0003_add_avatar_url.py conflicts with 0003_add_insight_meta_cache.py that already exists on main. Alembic will explode on upgrade. Needs to be renumbered to 0004_add_avatar_url.py with down_revision = "0003"

File: observal-server/api/routes/auth.py [hunk 2]
```diff
@@ -682,3 +684,88 @@ async def create_hooks_token(current_user: User = Depends(get_current_user)):
         detail="Hooks token created (30-day)",
     )
     return {"access_token": token, "expires_in": expires_in}
+
+
+# ── Avatar Upload ─────────────────────────────────────────────────
+
+_MAX_AVATAR_BYTES = 2 * 1024 * 1024
+_MAX_AVATAR_DATA_URL_LEN = 3 * 1024 * 1024
```
Comment (Haz3-jolt) [inline observal-server/api/routes/auth.py]: mismatch is confusing, 3mb here but 5mb in frontend? why?

File: observal-server/api/routes/auth.py [hunk 3]
```diff
@@ -682,3 +684,88 @@ async def create_hooks_token(current_user: User = Depends(get_current_user)):
...
+        raise HTTPException(status_code=422, detail="Image too large (max 2MB)")
+
+    signatures = _AVATAR_MAGIC_BYTES.get(mime_type, [])
+    if not any(raw.startswith(sig) for sig in signatures):
+        raise HTTPException(status_code=422, detail="File content does not match declared type")
+    if mime_type == "image/webp" and raw[8:12] != b"WEBP":
+        raise HTTPException(status_code=422, detail="File content does not match declared type")
+
+
+@router.put("/profile/avatar", response_model=UserResponse)
```
Comment (Haz3-jolt) [inline observal-server/api/routes/auth.py:732]: No rate limiting? do something like 10/min

**Review by Haz3-jolt — CHANGES_REQUESTED** [review]
0003_add_avatar_url.py conflicts with 0003_add_insight_meta_cache.py that already exists on main. Alembic will explode on upgrade. Needs to be renumbered to 0004_add_avatar_url.py with down_revision = "0003"

**Review by Haz3-jolt — APPROVED** [review]
